kolibri/logger/models.py

```diff
@@ -30,45 +33,56 @@ class BaseLogModel(models.Model):
         )
     )
 
-    user = models.ForeignKey(FacilityUser, blank=True, null=True, db_index=True)
-
     class Meta:
         abstract = True
 
+    def infer_dataset(self):
+        if self.user:
+            return self.user.dataset
+        else:
+            facility = Facility.get_default_facility()
+            assert facility, "Before you can save logs, you must have a facility"
+            return facility.dataset
 
-class ContentInteractionLog(BaseLogModel):
+
+class ContentSessionLog(BaseLogModel):
     """
-    This model provides a record of an interaction with a content item.
+    This model provides a record of interactions with a content item within a single visit to that content page.
     """
+    user = models.ForeignKey(FacilityUser, blank=True, null=True)
     content_id = UUIDField(db_index=True)
-    channel_id = UUIDField(db_index=True)
+    channel_id = UUIDField()
     start_timestamp = models.DateTimeField(auto_now_add=True)
-    completion_timestamp = models.DateTimeField(blank=True, null=True)
-    item_session = UUIDField()
-    kind = models.CharField(max_length=200, blank=True)  # indicates how extra_fields should be interpreted
-    extra_fields = models.TextField(blank=True)
+    end_timestamp = models.DateTimeField(blank=True, null=True)
+    time_spent = models.FloatField(help_text="(in seconds)", default=0.0)
+    kind = models.CharField(max_length=200)
+    extra_fields = models.TextField(default="{}")
 
 
 class ContentSummaryLog(BaseLogModel):
     """
-    This model provides a summary of all interactions of a user with a content item.
+    This model provides a summary of all interactions a user has had with a content item.
     """
+    user = models.ForeignKey(FacilityUser)
     content_id = UUIDField(db_index=True)
-    channel_id = UUIDField(db_index=True)
+    channel_id = UUIDField()
     start_timestamp = models.DateTimeField(auto_now_add=True)
-    last_activity_timestamp = models.DateTimeField(blank=True, null=True)
+    end_timestamp = models.DateTimeField(blank=True, null=True)
```

Comment:
Not sure I understand the change in nomenclature here, `last_activity_timestamp` is far more indicative of this, whereas `end_timestamp` tends to indicate that this log is no longer actively being used for logging.